src/pages/login/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import Button from '../../components/button';
+import ErrorMessage from '../../components/message';
+import { twMerge } from 'tailwind-merge';
+
+const Login = () => {
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [errors, setErrors] = useState<{ email?: string; password?: string }>({
+    email: '',
+    password: '',
+  });
+
+  const validateForm = () => {
+    const newErrors: { email?: string; password?: string } = {};
+
+    if (!email) {
+      newErrors.email = 'E-mail é obrigatório.';
+    } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
```

Comment:
Olha, é uma boa prática manter a facilidade de leitura do seu código. 
Esse seu regex nesse if não tá tão legal, não seria mais agradável algo como:

```suggestion
      const validateForm = () => {
    const newErrors: { email?: string; password?: string } = {};
    const isValidEmailFormat = /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
    
    if (!email) {
      newErrors.email = 'E-mail é obrigatório.';
    } else if (!isValidEmailFormat) {
```

Não é sempre só você que vai codar em um local, é importante deixar as coisas mais fáceis para os demais.

Reply:
Alteração feita, valeu!